File: day2/day2.py
import fileinput
import re
import string

part1, part2 = 0, 0
lines = list(fileinput.input("inputs/02.txt"))
for line in lines:
    minmax = re.findall(r'\d+', line)
    password = re.findall('[a-z]+', line)
    count = password[1].count(password[0])
    if  int(minmax[1]) >= count >= int(minmax[0]):
        part1 += 1
    pos1 = password[1][int(minmax[0])-1] == password[0]
    pos2 = password[1][int(minmax[1])-1] == password[0]
    if pos1 ^ pos2: #(pos1 and not pos2) or (pos2 and not pos1):
        part2 += 1


print("Part1:", part1)
print("Part2:", part2)

File: day2/day2.go
package main

import (
	"bytes"
	"fmt"
	"io/ioutil"
	"regexp"
	"strconv"
)

func day2(input []byte) int {

	var goodPass int
	reNum := regexp.MustCompile(`\d+`)
	rePass := regexp.MustCompile(`[a-z]+`)

	inArray := bytes.Split(input, []byte("\n"))
	for _, value := range inArray {
		minmax := reNum.FindAll(value, -1)

		password := rePass.FindAll(value, -1)
		count := bytes.Count(password[1], password[0])

		if count >= castBtoI(minmax[0]) && count <= castBtoI(minmax[1]) {
			goodPass++
		}
	}
	return goodPass
}

func castBtoI(byteSlice []byte) int {
	byteToInt, err := strconv.Atoi(string(byteSlice))
	check(err)
	return byteToInt
}

func check(e error) {
	if e != nil {
		panic(e)
	}
}

func main() {
	dat, err := ioutil.ReadFile("inputs/02.txt")
	check(err)
	answer := day2(dat)
	fmt.Println(answer)

	// EXAMPLE CASE(S)
	// answer := day2([]byte(
	// 	`1-3 a: abcde
	// 	1-3 b: cdefg
	// 	2-9 c: ccccccccc`))
	// fmt.Println(answer)

}
